Lesson5/Task1/RationalNumber.cs

```diff
@@ -0,0 +1,194 @@
+﻿public class RationalNumber : ICloneable
+{
+    /// <summary>
+    /// числитель
+    /// </summary>
+    private int _numerator = 0;
+    /// <summary>
+    /// знаменатель
+    /// </summary>
+    private int _denominator = 1;
+    /// <summary>
+    /// числитель
+    /// </summary>
+    public int Numerator { get => _numerator; }
+    /// <summary>
+    /// знаменатель
+    /// </summary>
+    public int Denominator { get => _denominator; }
+
+    /// <summary>
+    /// установка числителя и знаменателя с упрощением 
+    /// </summary>
+    /// <param name="numerator">числитель</param>
+    /// <param name="denominator">знаменатель</param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentOutOfRangeException"></exception>
+    public RationalNumber Set(int numerator, int denominator)
+    {
+        if (denominator == 0)
+            throw new ArgumentOutOfRangeException("Знаменатель равен 0!");
+        if (denominator < 0)
+        {//знак будет только у числителя!
+            numerator = -numerator;
+            denominator = -denominator;
+        }
+        _numerator = numerator;
+        _denominator = denominator;
+
+        return Simplify();
+    }
+
+    #region Конструкторы
+    public RationalNumber(int numerator, int denominator) => Set(numerator, denominator);
+    public RationalNumber() : this(0, 1) { }
+    #endregion
+
+    /// <summary>
+    /// Упрощение рационального числа
+    /// </summary>
+    /// <returns></returns>
+    private RationalNumber Simplify()
+    {
+        int nod = NOD(_numerator, _denominator);
+        _numerator /= nod;
+        _denominator /= nod;
+        return this;
```

Comment:
```suggestion
        return new(_numerator / nod, _denominator  / nod);
```
Любая операция - это новая дробь.